nrf_rpc/nrf_rpc.c

```diff
@@ -1144,20 +1162,66 @@ int nrf_rpc_init(nrf_rpc_err_handler_t err_handler)
 		return err;
 	}
 
-	for (i = 0; i < CONFIG_NRF_RPC_CMD_CTX_POOL_SIZE; i++) {
+	for (size_t i = 0; i < CONFIG_NRF_RPC_CMD_CTX_POOL_SIZE; i++) {
 		cmd_ctx_pool[i].id = i;
 		err = nrf_rpc_os_msg_init(&cmd_ctx_pool[i].recv_msg);
 		if (err < 0) {
 			return err;
 		}
 	}
 
-	err = transport_init(receive_handler);
+	global_err_handler = default_err_handler;
+
+	is_initialized = true;
+
+	return 0;
+}
+
+int nrf_rpc_init_group(const struct nrf_rpc_group *group)
+{
+	int err = nrf_rpc_prepare_init();
+	if (err < 0) {
+		return err;
+	}
+
+	err = transport_init_single(receive_handler, group);
+	if (err < 0) {
+		return err;
+	}
+
+	NRF_RPC_ASSERT(group_count > 0);
+	if (initialized_group_count == group_count - 1) {
```

Comment:
Shouldn't you use `waiting_group_count`?  If there are groups without `NRF_RPC_FLAGS_WAIT_ON_INIT` flag, the wait below will hang indefinitely. By the way, it's non-obvious that the function waits only if there's only one group that has not been initialized so far, so this should be documented or there should be a follow up filed to make the group initialization less coupled (so that each group can only wait for its own initialization completion).

Reply:
What I was trying to do here to here is to call the event_wait function only for the last group which was getting initialized. The function `groups_init_event_wait` checks for waiting threads so it will actually not hang. But on the other hand now that I look at it again there is a bug in this indeed. The `initialized_group_count` only increases for the threads which have the `NRF_RPC_FLAGS_WAIT_ON_INIT` flag enabled, so if there is one thread that doesn't then the wait will never get called. I will have to refactor this.